org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/jdt/JavaDebugServer.java

```diff
@@ -99,14 +97,20 @@ private void closeServerSocket() {
         serverSocket = null;
     }
 
-    private void closeConnection() {
-        if (connection != null) {
+    private void createConnectionThread(Socket connection) {
+        new Thread(() -> {
```

Comment:
fixed. will use Java Thread Pool to manage the newly created thread. and stop all threads when invoking stop().